ni/measurementlink/discovery/v1/discovery_service.proto

```diff
@@ -176,4 +180,29 @@ message EnumerateComputeNodesRequest {
 message EnumerateComputeNodesResponse {
   // The list of compute nodes that have registered themselves in the current session.
   repeated ComputeNodeDescriptor compute_nodes = 1;
+}
+
+message ResolveServiceWithInformationRequest {
+  // Required. This corresponds to the gRPC Full Name of the service and should match the information
+  // that was supplied in the RegisterServiceRequest message.
+  string provided_interface = 1;
+
+  // Optional. The service "class" that should be matched. If the value of this field is not specified and there
+  // is more than one matching service registered, an error is returned.
+  string service_class = 2;
+
+  // Optional. Indicates the deployment target from which the service should be resolved.
+  // The value of this field can be obtained from the results of the EnumerateComputeNodes method.
+  // If the value of this field is not specified, the service will be resolved from the
+  // local deployment target.  If the service cannot be resolved from the specified deployment
+  // target, an error is returned.
+  string deployment_target = 3;
+}
+
+message ResolveServiceWithInformationResponse {
+  // The canonical location information for the service.
+  ServiceLocation service_location = 1;
+
+  // The description of the service.
+  ServiceDescriptor service_descriptor = 2;
```

Comment:
Do we need to return all these details in `ServiceDescriptor`?

Reply:
No needed! Thus, I have included only string description alone.